### PR TITLE
VYZN: Navigation elements by their IFC type

### LINKED ISSUE #620: VYZN: Viewer: Hiding of one or more elements by TYPE by the user
This issue has been split-off from #384.

As a user I want to navigate elements by type (walls, slabs, ...). 

There should be a secondary tree-view that allows navigating by types.

Example from BimCollab:

202474317-0c782c31-6189-4438-8281-7178bc8c0522

### PR DESCRIPTION
**Pull request by Ibrahim5aad**
This PR adds a navigation mode that groups element by their IFC type. Closes #620  as it adds the required modification for the IfcIsolator to work with aggregated IFC elements by their type.

### CODE REVIEW

File: src/Components/BranchesControl.jsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ export default function Branches() {
       {branches.length > 1 && modelPath.repo !== undefined &&
         <Paper elevation={0} variant='control'
           sx={{
-            marginTop: '34px',
```
Comment (oo-bldrs) [inline src/Components/BranchesControl.jsx:79]: @Ibrahim5aad @OlegMoshkovich Is the change in margin due to how the page renders on different devices? Is this a one-off change or do we need breakpoints to handle different screen sizes?

File: src/Components/HideIcon.jsx
Comment (oo-bldrs) [inline src/Components/HideIcon.jsx:1]: Traditionally, other components in the project that are named `...Icon` are expected to be pure icons with no stateful logic. This component should carry a different name.

File: src/Components/HideIcon.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
```
Comment (oo-bldrs) [inline src/Components/HideIcon.jsx:2]: The project is based on Material UI which already comes with its own React Icon component and its own set of icons (https://mui.com/material-ui/icons/).

Adding Fort Awesome as a new dependency solely for icons is redundant.

(I am aware that this dependency was introduced in a different commit)

/cc @pablo-mayrgundter

File: src/Components/BranchesControl.jsx (hunk 1 above)
Comment (Ibrahim5aad) [reply src/Components/BranchesControl.jsx:79]: @oo-bldrs Nope, this just visual alignment I though it is convenient after I introduced another component.  It this exact value serves another purposes I'm not aware of I can revert this.

File: src/Components/HideIcon.jsx
Comment (Ibrahim5aad) [reply src/Components/HideIcon.jsx:1]: @oo-bldrs  What do you suggest? I'm not aware of React naming conventions I'm afraid.

Comment (pablo-mayrgundter) [reply src/Components/HideIcon.jsx:1]: For buttons with icons that have extended behavior, we've been using ControlButton, so e.g. HideControl.

File: src/Components/BranchesControl.jsx (hunk 1 above)
Comment (pablo-mayrgundter) [reply src/Components/BranchesControl.jsx:79]: See my overall comment about moving this into the NavTree.  Maybe the alignment will change.

File: src/Components/HideIcon.jsx
Comment (oo-bldrs) [reply src/Components/HideIcon.jsx:1]: `SomethingToggle` is preferred as that implies both: 1) this control has state, and 2) there are only 2 valid states.

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
One top-level design comment and also chimed in on a couple of the threads.

**Comment by Ibrahim5aad** [comment]
@pablo-mayrgundter Thanks for the feedback, will work on them and ping you.

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Hi Ibrahim,

Sorry for the slow review!  Last week turned into a travel week and I've been backlogged.

Please put this feature behind a feature flag e.g. `?features=navtypes` and we can take the rest of the design issues.

Ping us to discuss how to implement this, but basically all of us are in the process of doing larger PRs that will use feature flags.  I expect we'll put them in src/FeatureFlags, and either Ron or I will have smth there soon to reference.

Thanks!
Pablo

**Comment by Ibrahim5aad** [comment]
> features

Hi @pablo-mayrgundter, sorry for the late reply, I was in a long Eid vacation. I'll be looking into this now. I noticed you using the hook 'useExistInFeature' to implement this right?

**Comment by pablo-mayrgundter** [comment]
> > features
> 
> Hi @pablo-mayrgundter, sorry for the late reply, I was in a long Eid vacation. I'll be looking into this now. I noticed you using the hook 'useExistInFeature' to implement this right?

Yep, that's right.. I created a wiki page to document usage.  Lmk what info you've been missing and we can add it here:

https://github.com/bldrs-ai/Share/wiki/Design:-URL-Structure:-Feature-Flags

**Comment by Adrian62D** [comment]
@pablo-mayrgundter This has already been implement as far as I see. You can use ?feature=navTypes in the live preview.

**Comment by Ibrahim5aad** [comment]
@pablo-mayrgundter It is already in place. You can give it a try.

**Review by pablo-mayrgundter — APPROVED** [review]

**Pull request merged**